### PR TITLE
fix(gleam): extract parameters for external functions

### LINKED ISSUE #1110: follow-up: extract parameters for Gleam external_function in both engines
Deferred from PR #1105 review.

**Original reviewer comment:** https://github.com/optave/ops-codegraph-tool/pull/1105#discussion_r3222098271

**Context**

`handle_external_function` (Rust) and `handleExternalFunction` (JS) both hardcode `children: None` / omit `children`, while their counterparts for regular `function` extract parameters as children. This creates a silent asymmetry between regular and external Gleam functions even though both have a full parameter list in their signatures.

**Why deferred**

The current PR's stated goal is to port the existing WASM/JS Gleam extractor to native Rust at parity. The native Rust implementation faithfully mirrors the existing WASM/JS behavior. Fixing this requires changes in BOTH engines plus corresponding unit tests, which goes beyond "port to Rust" scope.

**Suggested fix**

- Add `extract_params` call to `handle_external_function` in `crates/codegraph-core/src/extractors/gleam.rs`.
- Mirror the change in `handleExternalFunction` in `src/extractors/gleam.ts` so the JS engine keeps parity.
- Add unit tests covering external function parameter extraction in both Rust and TypeScript test suites.
- Verify the Gleam tree-sitter grammar exposes a `parameters` field on `external_function` nodes (or fall back to `function_parameters` child).

### PR DESCRIPTION
**Pull request by carlos-alm**
## Summary

- Mirror regular `function` behavior in `handle_external_function` (Rust) and `handleExternalFunction` (JS) so external function definitions expose their parameter list as children in both engines.
- Tree-sitter `external_function` exposes a `parameters` field (aliased from `external_function_parameters` to `function_parameters`) with the same shape as regular `function`, so `extract_params`/`extractParams` can be reused as-is. Type-only external params (no `name` field) are correctly omitted.

## Test plan

- [x] Rust: `cargo test --lib extractors::gleam` — 12/12 pass (2 new tests cover named and nameless external params).
- [x] JS: `npx vitest run tests/parsers/gleam.test.ts` — 6/6 pass (2 new tests cover named and nameless external params).
- [x] JS: `npx vitest run tests/parsers/` — 476/476 pass after `npm run build`.
- [x] Rust: `cargo test --lib` — 299/299 pass.
- [x] End-to-end smoke (`--engine wasm`): `codegraph children parse` lists `input` and `base` as `parameter` children.

Closes #1110

### CODE REVIEW

File: tests/parsers/gleam.test.ts [hunk 1]
```diff
@@ -45,4 +45,25 @@ import gleam/string`);
...
+    expect(names).toContain('base');
+    expect(parseFn?.children?.every((c) => c.kind === 'parameter')).toBe(true);
+  });
+
+  it('omits children for external functions with type-only parameters', () => {
+    const symbols = parseGleam(`pub external fn random() -> Int = "rand" "uniform"`);
+    const randomFn = symbols.definitions.find((d) => d.name === 'random');
+    expect(randomFn).toBeDefined();
+    expect(randomFn?.children).toBeUndefined();
+  });
```
[the remark this answers is not in the document]
Comment (carlos-alm) [reply tests/parsers/gleam.test.ts:70]: Fixed in 7dde2c3 — both the JS and Rust tests now use `random(Int, String)` (type-only parameters with no `name` field) instead of `random()` (zero parameters), so the assertion that `children` is undefined actually exercises the documented edge case: parameter nodes that exist in the tree but get filtered out by `extractParams`/`extract_params` because they lack a `name` field.

**Comment by carlos-alm** [comment]
@greptileai
